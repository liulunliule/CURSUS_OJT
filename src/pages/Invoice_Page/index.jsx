import React, { useEffect } from "react";
import "./index.scss";
import { Link } from "react-router-dom";
import { Logo_dark } from "../../assets";
import { useDispatch, useSelector } from "react-redux";
import { fetchAddr } from "../../redux/features/checkoutSlice";
import { fetchShoppingCart } from "../../redux/features/shoppingCartSlice";

const Invoice_Page = () => {
  const dispatch = useDispatch();
  const account = useSelector((state) => state.user.account);
  const userId = account.id;
  const { addr, order, status, error } = useSelector((state) => state.checkout);
  const items = useSelector((state) => state.shoppingCart.items);
  const parsePrice = (price) => parseFloat(price);

  const calculateDiscount = (price) => (parsePrice(price) * 0.2).toFixed(2);
  const calculateTotalAmount = (price) =>
    (parsePrice(price) + parseFloat(calculateDiscount(price))).toFixed(2);
  const totalDiscount = items
    .reduce((sum, item) => sum + parseFloat(calculateDiscount(item.price)), 0)
    .toFixed(2);
  const totalAmount = items
    .reduce(
      (sum, item) => sum + parseFloat(calculateTotalAmount(item.price)),
      0
    )
    .toFixed(2);

  useEffect(() => {
    dispatch(fetchShoppingCart(userId));
    dispatch(fetchAddr(userId));
  }, [dispatch]);
  return (
    <div className="invoice">
      <div className="invoice_header ">
        <div className="container">
          <div className="row justify-content-md-center">
            <div className="col-md-8">
              <div className="invoice_header_main">
                <div className="invoice_header_item">
                  <div className="invoice_logo">
                    <Link to="/">
                      <img src={Logo_dark} alt="" />
                    </Link>
                  </div>
                  <p>Invoice</p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="invoice_main_body">
        <div class="container">
          <div class="row justify-content-md-center">
            <div class="col-md-8">
              <div class="invoice_body">
                <div class="invoice_date_info">
                  <ul>
                    <li>
                      <div class="invoice_date_info_list">
                        <span>Date :</span>
                        10 April 2020
                      </div>
                    </li>
                    <li>
                      <div class="invoice_date_info_list">
                        <span>Invoice No :</span>IVIP12023598
                      </div>
                    </li>
                    <li>
                      <div class="invoice_date_info_list">
                        <span>Order ID :</span>
                        1258963487
                      </div>
                    </li>
                  </ul>
                </div>
                <div class="invoice_dts">
                  <div class="row">
                    <div class="col-md-12">
                      <h2 class="invoice_title">Invoice</h2>
                    </div>
                    <div class="col-sm-6">
                      <div class="invoice_title_item">
                        <h4>To</h4>
                        <ul>
                          <li>
                            <div class="invoice_date_info_list">
                              {addr.userName}
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              {addr.Address1}
                              <br /> {addr.Address2}
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              {addr.City}
                              <br />
                              {addr.State}
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              {addr.ZipCode}
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              {addr.Country}
                            </div>
                          </li>
                        </ul>
                      </div>
                    </div>
                    <div class="col-sm-6">
                      <div class="invoice_title_item">
                        <h4>Cursus</h4>
                        <ul>
                          <li>
                            <div class="invoice_date_info_list">Cursus LTD</div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              #1234, Shahid karnail Singh Nagar,
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">
                              Near MBD Mall,
                            </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">141001</div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">Ludhiana </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">Punjab </div>
                          </li>
                          <li>
                            <div class="invoice_date_info_list">India </div>
                          </li>
                        </ul>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="invoice_table">
                  <div>
                    <table class="table">
                      <thead>
                        <tr>
                          <th scope="col">Item</th>
                          <th scope="col">Price</th>
                          <th scope="col">Discount</th>
                          <th scope="col">Total Amount</th>
                        </tr>
                      </thead>
                      <tbody>
                        {items.map((item) => (
                          <tr>
                            <th scope="row">
                              <div class="user_dt_trans">
                                <p>{item.titilecourse}</p>
                              </div>
                            </th>
                            <td>
                              <div class="user_dt_trans">
                                <p>${parsePrice(item.price).toFixed(2)}</p>
                              </div>
                            </td>
                            <td>
                              <div class="user_dt_trans">
                                <p>${calculateDiscount(item.price)}</p>
                              </div>
                            </td>
                            <td>
                              <div class="user_dt_trans">
                                <p>${calculateTotalAmount(item.price)}</p>
                              </div>
                            </td>
                          </tr>
                        ))}
                        <tr>
                          <td colspan="1"></td>
                          <td colspan="3">
                            <div class="user_dt_trans total">
                              <div class="invoice_total">
                                Invoice Total : USD ${totalAmount}
                              </div>
                              <p>Total Discount : USD ${totalDiscount}</p>
                              <p>Paid via Paypal</p>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
                <div class="invoice_footer">
                  <div class="leftfooter">
                    <p>Thanks for buying.</p>
                  </div>
                  <div class="righttfooter">
                    <a class="print_btn" href="javascript:window.print();">
                      Print
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Invoice_Page;
