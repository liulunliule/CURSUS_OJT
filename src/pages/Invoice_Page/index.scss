.invoice{
    font-family: 'Roboto', sans-serif;
    background-color: #f7f7f7 !important;
    padding: 0;
    .container{
        max-width: 1200px;
    }
    ul {
        list-style: none;
        margin-bottom: 0 !important;
        padding-left: 0 !important;
    }
    &_header {
        position: relative !important;
        z-index: 1 !important;
        width: 100% !important;
        height: 60px !important;
        background: transparent !important;
        display: flex !important;
        align-items: center !important;
        box-shadow: none !important;
    
        &_main{
            background: #333;
            float: left;
            width: 100%;
            height: 60px;
            display: flex;
            align-items: center;
            padding: 30px;
        }
        &_item {
            float: left;
            width: 100%;
            // text-align: center;
            margin-top: 10px;
            p {
                font-size: 20px;
                font-family: 'Roboto', sans-serif;
                font-weight: 500;
                color: #fff;
                padding: 5px 0;
                float: right;
            }
        }
    }
    &_logo {
        float: left;
        width: 135px;
    }
    &_body {
        padding: 50px 30px;
        background: #fff;
        float: left;
        width: 100%;
    }
    &_date_info {
        float: right;
        width: 100%;
        ul {
            float: right;
            list-style: none;
            margin-bottom: 0;
            padding-left: 0;
            li {
                display: block;
                margin-bottom: 5px;
            }
        }
        &_list {
            font-size: 14px;
            font-weight: 400;
            font-family: 'Roboto', sans-serif;
            color: #686f7a;
            text-align: left;
            line-height: 24px;

            span {
                color: #333;
                margin-right: 10px;
            }
        }
    }
    &_dts {
        float: left;
        width: 100%;
        margin-top: 35px;
    }
    &_title {
        float: left;
        width: 100%;
        margin-top: 0;
        margin-bottom: 0;
        font-size: 24px;
        font-weight: 600;
        color: #333;
        text-align: left;
        &_item{
            float: left;
            width: 100%;
            margin-top: 30px;

            ul li {
                display: block;
                margin-bottom: 5px;
            }

            h4 {
                font-size: 18px;
                font-weight: 500;
                font-family: 'Roboto', sans-serif;
                color: #333;
                text-align: left;
                margin-bottom: 15px;
            }
        }
    }
    &_table {
        float: left;
        width: 100%;
        margin-top: 50px;
        background: #fafafa;
        border-radius: 5px;
        .table {
            margin-bottom: 0 !important;

            thead th {
                vertical-align: bottom;
                border-bottom: 1px solid #efefef !important;
                font-size: 14px;
                font-weight: 500;
                color: #333;
                font-family: 'Roboto', sans-serif;
                padding: 15px 20px;
            }
            th {
                padding: 20px !important;
                vertical-align: top !important;
                border-top: 0 !important;
            }
            td {
                border-top: 1px solid #efefef !important;
                padding: 20px !important;
                vertical-align: top !important;
            }
        }
    }
    &_footer {
        float: left;
        width: 100%;
        margin-top: 30px;
        .leftfooter {
            float: left;
        }
        .righttfooter {
            float: right;
        }
        .print_btn {
            float: right;
            height: 40px;
            background: #ed2a26;
            padding: 11px 30px;
            color: #fff;
            font-weight: 500;
            font-size: 14px;
            text-align: center;
            text-transform: uppercase;
            border-radius: 3px;
        }
    }
    .total{
        text-align: right;
        .invoice_total{
            font-size: 16px;
            font-weight: 500;
            font-family: 'Roboto', sans-serif;
            color: #242424;
            margin-bottom: 0px;
            margin-bottom: 10px;
        }
    }
}
.user_dt_trans {
    font-size: 14px;
    font-weight: 400;
    color: #686f7a;
    text-align: left;
    line-height: 24px;
}

