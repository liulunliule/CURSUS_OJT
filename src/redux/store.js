import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
