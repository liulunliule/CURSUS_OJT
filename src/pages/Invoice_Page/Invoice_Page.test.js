import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Invoice_Page from '.';

test('test Invoice Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Invoice_Page/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
