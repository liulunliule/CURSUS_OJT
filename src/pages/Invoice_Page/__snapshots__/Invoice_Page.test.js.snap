// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Invoice Page 1`] = `
<div
  className="invoice"
>
  <div
    className="invoice_header "
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-md-center"
      >
        <div
          className="col-md-8"
        >
          <div
            className="invoice_header_main"
          >
            <div
              className="invoice_header_item"
            >
              <div
                className="invoice_logo"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    src="logo_dark.svg"
                  />
                </a>
              </div>
              <p>
                Invoice
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="invoice_main_body"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-md-center"
      >
        <div
          class="col-md-8"
        >
          <div
            class="invoice_body"
          >
            <div
              class="invoice_date_info"
            >
              <ul>
                <li>
                  <div
                    class="invoice_date_info_list"
                  >
                    <span>
                      Date :
                    </span>
                    10 April 2020
                  </div>
                </li>
                <li>
                  <div
                    class="invoice_date_info_list"
                  >
                    <span>
                      Invoice No :
                    </span>
                    IVIP12023598
                  </div>
                </li>
                <li>
                  <div
                    class="invoice_date_info_list"
                  >
                    <span>
                      Order ID :
                    </span>
                    1258963487
                  </div>
                </li>
              </ul>
            </div>
            <div
              class="invoice_dts"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-12"
                >
                  <h2
                    class="invoice_title"
                  >
                    Invoice
                  </h2>
                </div>
                <div
                  class="col-sm-6"
                >
                  <div
                    class="invoice_title_item"
                  >
                    <h4>
                      To
                    </h4>
                    <ul>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        />
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          <br />
                           
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          <br />
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        />
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        />
                      </li>
                    </ul>
                  </div>
                </div>
                <div
                  class="col-sm-6"
                >
                  <div
                    class="invoice_title_item"
                  >
                    <h4>
                      Cursus
                    </h4>
                    <ul>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          Cursus LTD
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          #1234, Shahid karnail Singh Nagar,
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          Near MBD Mall,
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          141001
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          Ludhiana 
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          Punjab 
                        </div>
                      </li>
                      <li>
                        <div
                          class="invoice_date_info_list"
                        >
                          India 
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="invoice_table"
            >
              <div>
                <table
                  class="table"
                >
                  <thead>
                    <tr>
                      <th
                        scope="col"
                      >
                        Item
                      </th>
                      <th
                        scope="col"
                      >
                        Price
                      </th>
                      <th
                        scope="col"
                      >
                        Discount
                      </th>
                      <th
                        scope="col"
                      >
                        Total Amount
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        colspan="1"
                      />
                      <td
                        colspan="3"
                      >
                        <div
                          class="user_dt_trans total"
                        >
                          <div
                            class="invoice_total"
                          >
                            Invoice Total : USD $
                            0.00
                          </div>
                          <p>
                            Total Discount : USD $
                            0.00
                          </p>
                          <p>
                            Paid via Paypal
                          </p>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <div
              class="invoice_footer"
            >
              <div
                class="leftfooter"
              >
                <p>
                  Thanks for buying.
                </p>
              </div>
              <div
                class="righttfooter"
              >
                <a
                  class="print_btn"
                  href="javascript:window.print();"
                >
                  Print
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
